src/main/hopur0/games/tictactoe/Game.java

```diff
@@ -1,7 +1,55 @@
 package hopur0.games.tictactoe;
 
 class Game {
-    public static void main(String[] args) {
-        System.out.println("what a man is Salah");
+  UI ui;
+  Board board;
+  Player[] players;
+
+  public Game(UI ui) {
+    this.ui = ui;
+    board = new Board();
+    initializePlayers();
+    gameLoop();
+  }
+
+  private void initializePlayers() {
+    String firstPlayerName = ui.getString("Enter first player name:");
+    String secondPlayerName = ui.getString("Enter second player name:");
+    players = new Player[] {
+      new Player(firstPlayerName, 'x'),
+      new Player(secondPlayerName, 'o')
+    };
+  }
+
+  private void gameLoop() {
+    int currentPlayer = 0;
+
+    while (board.winner() == null && !board.full()) {
+      ui.showMessage("----------------------------");
+      ui.showMessage("Current state:" + System.lineSeparator() + board.toString());
+      ui.showMessage(players[currentPlayer].getName() + " plays:");
+
+      int selectedTile = 0;
+      do {
+        selectedTile = ui.getInt("Enter tile number:");
+      } while (!board.validTile(selectedTile) || board.getTile(selectedTile) != null);
+
+      board.setTile(players[currentPlayer], selectedTile);
+      currentPlayer = (currentPlayer + 1) % 2;
     }
+
+    ui.showMessage("----------------------------");
+    ui.showMessage("Final state:" + System.lineSeparator() + board.toString());
+
+    if (board.full()) {
```

Comment:
If a player wins and fills the board at the same time would't that cause the code to display a draw?